app/models/chargeback/consumption_history.rb

```diff
@@ -43,5 +46,29 @@ def self.base_rollup_scope
     end
 
     private_class_method :base_rollup_scope
+
+    def self.uniq_timestamp_record_map(report_scope, group_by_tenant = false)
+      main_select = MetricRollup.select(:id, :resource_id).arel.ast.to_sql
+                                .gsub("SELECT", "DISTINCT ON (resource_type, resource_id, timestamp)")
```

Comment:
have you tried
`.distinct(:resource_type, :resource_id, :timestamp)`

I'd almost prefer raw sql over sql generation and munging.